default.nix

```diff
@@ -604,9 +605,9 @@ let overrideCabal = pkg: f: if pkg == null then null else lib.overrideCabal pkg
 in let this = rec {
   overrideForGhcjs = haskellPackages: haskellPackages.override {
     overrides = self: super: {
-      mkDerivation = drv: super.mkDerivation.override {
-        hscolour = ghc.hscolour;
-      } (drv // {
+      hscolour = ghc.hscolour;
```

Comment:
To be honest I don't know what "we want." ;) But what you're saying seems to make sense. I don't know what happened to the `override` attribute on `mkDerivation` since this was written but I don't see an easy way to recreate that behavior. It's quite possible I'm missing something. https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/haskell-modules/default.nix#L15-L41